functions/src/chat.ts

```diff
@@ -19,44 +20,50 @@ export async function deleteLobbyChatMessages(lobbyDoc: DocumentReference<Lobby>
   await batch.commit();
 }
 
-export async function deleteChatRooms(lobbyData: Lobby, lobbyDoc: DocumentReference<Lobby>, transaction: Transaction) {
+export async function deleteChatCollections(lobbyDoc: firestore.DocumentReference<Lobby>) {
+  const batch = db.batch();
+
+  const chatRooms = await getChatRoomCollection(lobbyDoc).get();
+
+  for (const chatRoomDoc of chatRooms.docs) {
+    // place messages in batch
+    const chatMessages = await getChatRoomMessagesCollection(chatRoomDoc.ref).get();
+    chatMessages.docs.map((messageDoc) => batch.delete(messageDoc.ref));
+
+    // place room in batch
+    batch.delete(chatRoomDoc.ref);
+  }
+
+  await batch.commit();
+}
+
+export async function setAnswers(lobbyData: Lobby, lobbyDoc: DocumentReference<Lobby>, transaction: Transaction) {
   const { players, uids } = lobbyData;
-  const chatRoomsSnapshot = await transaction.get(getChatRoomCollection(lobbyDoc));
-  const chatRooms = chatRoomsSnapshot.docs.map((room) => room.ref);
-
-  const chatMessages = await Promise.all(
-    chatRooms.map(async (room) => {
-      const chatMessagesSnapshot = await transaction.get(getChatRoomMessagesCollection(room));
-      return chatMessagesSnapshot.docs;
-    })
-  );
+  const promptAnswersSnaphot = await transaction.get(getPromptAnswerCollection(lobbyDoc));
 
   // delete stalker
   const stalkers = await transaction.get(getPrivatePlayerCollection(lobbyDoc).where("stalker", "==", true));
 
-  for (const messageDoc of chatMessages.flat()) {
-    // get message info
-    const message = messageDoc.data();
-    // if the message is the prompt answer then add it to their player object
-    if (message.isPromptAnswer) {
-      // get index and add their prompt answer to their player object
-      const senderIndex = uids.indexOf(message.sender);
-      players[senderIndex].promptAnswer = message.text;
-    }
-    // delete all messages
-    transaction.delete(messageDoc.ref);
+  for (const stalker of stalkers.docs) {
+    transaction.update(stalker.ref, { stalker: false });
   }
 
-  for (const room of chatRooms) {
-    transaction.delete(room);
-  }
+  // iterate through promptAnswer docs and place the answer on their player object
+  for (const promptAnswerDoc of promptAnswersSnaphot.docs) {
+    const promptData = promptAnswerDoc.data();
+    let promptAns = promptData.answer;
 
-  for (const stalker of stalkers.docs) {
-    transaction.update(stalker.ref, { stalker: false });
+    if (promptAns === "" || !promptAnswerDoc.exists) {
```

Comment:
don't need to check `!promptAnswerDoc.exists` because querries already garentee that the documents exist